distrobuilder/main.go

```diff
@@ -230,6 +230,90 @@ func (c *cmdGlobal) preRunBuild(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	var mounts []shared.ChrootMount
+	env := c.definition.Environment
+
+	isVM, err := cmd.Flags().GetBool("vm")
+	if err != nil {
+		return err
+	}
+
+	if isVM {
+		imgFilename, err := shared.RenderTemplate(fmt.Sprintf("%s.raw", c.definition.Image.Name), c.definition)
+		if err != nil {
+			return err
+		}
+
+		imgFile := filepath.Join(c.flagCacheDir, imgFilename)
+
+		// TODO: (monstermunchkin) Get image size from flag.
+		err = createEmptyDiskImage(imgFile, 2361393152)
+		if err != nil {
+			return err
+		}
+
+		err = createPartitions(imgFile)
+		if err != nil {
+			return err
+		}
+
+		err = mountImage(imgFile)
+		if err != nil {
+			return err
+		}
+		defer umountImage()
+
+		err = createRootFS()
+		if err != nil {
+			return err
+		}
+
+		err = mountRootPartition(c.sourceDir)
+		if err != nil {
+			return err
+		}
+		defer lxd.RunCommand("umount", "-R", c.sourceDir)
+
+		err = createUEFIFS()
+		if err != nil {
+			return err
+		}
+
+		err = mountUEFIPartition(c.sourceDir)
+		if err != nil {
+			return err
+		}
+
+		partUUID, err := getRootfsPartitionUUID()
```

Comment:
To generate the fstab, we'll actually need the `blkid` for both the UEFI and root partitions.